packages/edit-post/src/components/fullscreen-mode/index.js

```diff
@@ -17,11 +18,15 @@ class FullscreenMode extends Component {
 
 	sync() {
 		const { isActive } = this.props;
+		let delay = 0;
 		if ( isActive ) {
 			document.body.classList.add( 'is-fullscreen-mode' );
+			delay = 300; // .is-fullscreen-mode has a fade in animation of 0.3s
 		} else {
 			document.body.classList.remove( 'is-fullscreen-mode' );
 		}
+
+		setTimeout( () => doAction( 'fullscreenModeToggled' ), delay );
```

Comment:
It's an interesting usage of action hooks. Combined with `recalculatePositionHook` prop it makes it nicely decoupled from `edit-post`. However, I'm wondering if we should programmatically trigger `resize` event on `window` object instead to simulate what in reality happens in here. @aduth or @youknowriad any thoughts on that?

Reply:
I personally think this is not the right way to go about it. We already have `refresh` functions in `Popover` and `DropDown` component to be able to trigger a refresh when necessary. I agree though that it's not usable here because there's no directly relation between the "fullsize toggle" and any other popover shown in the screen.

My idea solution would be to drop this `refresh` method and this hook entirely and make sure the Popover updates itself when the position of its wrapper node change or its content is resized. I know this is not achievable easily as there's no event to detect such changes but I think we should at least try a `setInterval` check (see if it's not impactful performance wise).

Thoughs?

Reply:
I would agree this is the wrong fix for the issue at hand, as it adds complexity. It also assumes specific editor interface configurations, that although they are relatively stable might change in the future in which case this implementation will need to be revisited.

I would suggest a copule of other possible approaches to fixing this issue:

1. Toggle off the tips when fullscreen mode is engaged — similar to pressing the X button. Essentially a click that isn't dismiss or next, in this case "fullscreen", feels like an implicit "I'm done with tips" click.
2. Toggle them off and on again.
3. Change how the tips work so they are either inline, or perhaps even truly modal — i.e. you HAVE to click next or dismiss them before you can proceed. 
4. Consider this more or less a non-issue. The tips aren't _that_ disconnected, and as soon as you press Next, the subsequent tooltip is correctly positioned.

Overall, I would imagine so many changes happening to tips in Phase 2 that I don't think it would be too healthy for the code to optimize for this particular configuration right at this point. 

Option 3 is my personal favorite at it might also have benefits for accessibility. More than that, there are a bunch of issues filed in the Tips label at the moment, that are related to the fact that you _can change content below the tips without dismissing them_. For example you can toggle on or off the settings tab multiple times while still reading the tip that suggests you toggle them on.

Reply:
Thanks for all the ideas! Yeah, I agree this can lead to an over-engineered solution, although I tried to make it as simple as possible (my initial attempt was dispatching actions for connecting `edit-post` with `nux`).

Before considering what approach we should consider for going ahead, let me summarize where this issue is coming from. 

Calypso is integrating the Gutenberg editor but there is a lot of work to do in order to have something stable. So, in the meantime, we're going to redirect the users to the WP Admin Gutenberg editor (Automattic/wp-calypso#27317) using Calypsoify.

Calypsoify changes the WP Admin UI so it looks like Calypso, and the way of doing that for the Gutenberg editor is by enabling the Fullscreen mode (the Calypso editor is pretty similar to the Gutenberg editor in Fullscreen mode).

This means that the Fullscreen mode will be enabled programmatically. The user is not doing anything so I don't think we can hide the tips or force the user to dismiss them.

That's why I'd go for the @gziolo or @youknowriad suggestions, but I'm also wondering if this not really an issue as @jasmussen pointed and it's very related to Calypsoify and we should fix it there.

Reply:
I personally would love a generic solution in `Popover` if possible because this refresh can be necessary in a wide range of situations.

Reply:
I'm thinking in using [`ResizeObserver`](https://alligator.io/js/resize-observer/) (and its [polyfill](https://github.com/que-etc/resize-observer-polyfill)) for detecting when the anchor of the popover changes, instead of using `setInterval`.

What are your thoughts about that?

Reply:
@mmtr Worth a try, I think we need both a resize and a position observer though, I don't know if it takes into account all the use-cases.

Reply:
Just to be clear though, this is not a Calypso-only fix. 🙂 
If folks toggle into fullscreen with the NUX (or any other popovers, for that matter) visible, the popover doesn't reposition.

I think this solution is still a worthwhile extension for the `Popover` component, but I have to agree the better one would be to observe the anchor position.

We already use `MutationObserver` in Gutenberg, so we might want to try listening with something roughly like this:

```es6
const observer = new MutationObserver( mutations => {
  if ( _.find( mutations, { type: 'attributes', attributeName: 'style' } ) ) {
    this.refresh();
  }
} );

observer.observe( this.anchorRef, { attributes: true } );
```

(I'm not super familiar with `MutationObserver`, maybe there's a better way to check if only the anchor position has changed, instead of any style?)

Also, I'd probably still only enable the observer if some sort of `repositionWhenAnchorRepositions` prop is true, to keep the existing behaviour untouched, and avoid performance issues.

Reply:
`MutationObserver` won't work since it only detects changes done in the DOM and we don't manipulate the DOM of the anchor. 

I'm investigating the new `ResizeObserver` and `IntersectionObserver` observers, which apparently detects changes in the sizes and positions respectively.

Reply:
@mmtr Ah yup, that makes sense 🙁